roles/openshift_health_checker/test/search_journalctl_test.py

```diff
@@ -0,0 +1,160 @@
+import pytest
+import search_journalctl
+
+
+def canned_search_journalctl(get_log_input=None):
+    """Create a search_journalctl object with canned get_log_input method"""
+    module = search_journalctl
+    if get_log_input:
+        module.get_log_input = get_log_input
+    return module
+
+
+DEFAULT_TIMESTAMP = 1496341364
+
+
+def get_timestamp(modifier=0):
+    return DEFAULT_TIMESTAMP + modifier
+
+
+def get_timestamp_microseconds(modifier=0):
+    return get_timestamp(modifier) * 1000000
+
+
+def create_test_log_object(stamp, msg):
+    return '{{"__REALTIME_TIMESTAMP": "{}", "MESSAGE": "{}"}}'.format(stamp, msg)
+
+
+@pytest.mark.parametrize('matchers,log_input,expected_matches,expected_errors', [
+    (
+        [
+            {
+                "start_regexp": r"Sample Logs Beginning",
+                "regexp": r"test log message",
+                "unit": "test",
+            },
+        ],
+        [
+            create_test_log_object(get_timestamp_microseconds(), "test log message"),
+            create_test_log_object(get_timestamp_microseconds(), "Sample Logs Beginning"),
+        ],
+        ["test log message"],
+        [],
+    ),
+    # test with invalid json in log input
+    (
+        [
+            {
+                "start_regexp": r"Sample Logs Beginning",
+                "regexp": r"test log message",
+                "unit": "test-unit",
+            },
+        ],
+        [
+            '{__REALTIME_TIMESTAMP: ' + str(get_timestamp_microseconds()) + ', "MESSAGE": "test log message"}',
+        ],
+        [],
+        ["invalid json", "test-unit", "test log message"],
+    ),
+    # test with invalid regexp
+    (
+        [
+            {
+                "start_regexp": r"Sample Logs Beginning",
+                "regexp": r"test [ log message",
+                "unit": "test",
+            },
+        ],
+        [
+            create_test_log_object(get_timestamp_microseconds(), "test log message"),
+            create_test_log_object(get_timestamp_microseconds(), "sample log message"),
+            create_test_log_object(get_timestamp_microseconds(), "fake log message"),
+            create_test_log_object(get_timestamp_microseconds(), "dummy log message"),
+            create_test_log_object(get_timestamp_microseconds(), "Sample Logs Beginning"),
+        ],
+        [],
+        ["invalid regular expression"],
+    ),
+])
+def test_get_log_matches(matchers, log_input, expected_matches, expected_errors):
+    def get_log_input(matcher):
+        return log_input
+
+    module = canned_search_journalctl(get_log_input)
+    matched_regexp, errors = module.get_log_matches(matchers, 500, 60 * 60)
+
+    assert set(matched_regexp) == set(expected_matches)
+    for partial_err_msg in expected_errors:
+        for actual_err_msg in errors:
+            assert partial_err_msg in actual_err_msg
```

Comment:
Thanks, addressed this. Errors are not duplicate, but in the case of this test, I was receiving a single one each time, so the test was not failing. Addressed this to handle cases where more than one error might be returned by `get_log_matches`